common/src/test/java/com/google/auto/common/SuperficialValidationTest.java

```diff
@@ -199,6 +199,42 @@ public void handlesRecursiveType() {
         .compilesWithoutError();
   }
 
+  @Test
+  public void handlesRecursiveSuperinterface() {
+    JavaFileObject javaFileObject = JavaFileObjects.forSourceLines(
+        "test.TestClass",
+        "package test;",
+        "",
+        "interface TestClass implements TestClass {}");
```

Comment:
Was this failing before this change? This seems like an outright error that should never call annotation processing

Reply:
This didn't fail before the change - just added coverage for this case since @eamonnmcmanus suggested doing so in #660. The test does validate that we don't reach annotation processing.